### PR TITLE
[Datasets] Cherry-pick docs changes and small UX fixes for Datasets GA.

### PR DESCRIPTION
**Pull request by clarkzinzow**
This PR contains all docs changes and small UX fixes for Datasets GA, submitted as a mono-PR since later changes stack upon earlier ones, so we won’t get a cohesive docs + code build without all of the relevant changes.

Although none of the UX code changes should affect any Datasets release tests, if release tests are already going to be rerun, rerunning the Datasets releases tests on these code changes should thoroughly de-risk any surprise pop-ups.

Each commit is the corresponding master commit. Each individual commit is tracked in the 1.13 release tracker: https://docs.google.com/spreadsheets/d/1hYOvDaHc2W-ta1cn42jrM4bywfCVtkCpvU-DSeYtIjE/edit#gid=0

## Checklist

- [ ] All Datasets tests pass in CI.
- [ ] Docs build passes and Datasets docs look as intended.
- [ ] (Optional) Release tests pass on resulting wheel.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
Just reviewed. The only pick I'm a little nervous about is this one:

```
Datasets: Explicit resource allocation option
```

This is since the interactions with other frameworks could be subtle and this code has a high likelihood of introducing a bug. However, given that it's been in master and passing tests for quite a while, I think this risk is somewhat mitigated. Not picking it might actually be the worse option. However if there are no conflicts we should consider not picking it.

@jianoaix I also noticed that "Creating Datasets" still is using the "Arrow Range" and "ray.data.range_arrow()" syntax instead of `range_table()`. Could we change this to "Tabular Range" and "range_table()"? Not sure how this passed the doc code tests, since range_arrow() should be raising an exception now.

**Comment by jianoaix** [comment]
> Arrow Range

Yep, I can make a fix.

**Comment by clarkzinzow** [comment]
@ericl Good catch on `range_arrow()`! And I had assumed that all of the code in `doc/source/data/doc_code` is getting run in CI, we should double-check that.

As for explicit resource allocation, I thought that the inclusion is fine since it preserves the existing behavior by default, and we have several users that are asking for it that want to stick to releases rather than nightlies. If we're real nervous about it, we can revert that commit.

Also I just pushed up a commit that should fix the docs build; we had some AIR cross-references in the FAQ that don't exist yet in the 1.13 branch.

**Comment by jianoaix** [comment]
It looks fine to cherrypick but not necessarily GA. Can we mark the explicit resource allocation feature as alpha?

**Comment by clarkzinzow** [comment]
@jianoaix I just noticed some typos for our resource allocation key concepts docs, pushing up a PR now with those fixes + a note about it being experimental.

**Comment by clarkzinzow** [comment]
@jianoaix @ericl https://github.com/ray-project/ray/pull/25038

File: doc/test_myst_doc.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import argparse
```
Comment (jianoaix) [inline doc/test_myst_doc.py:5]: This file was added from https://github.com/ray-project/ray/commit/c74886a55e81dc8f4150094bfec69d875f0b365d to fix CI test for notebooks.

File: bazel/python.bzl [hunk 2]
```diff
@@ -1,3 +1,5 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
```
Comment (jianoaix) [inline bazel/python.bzl:1]: This file was added from https://github.com/ray-project/ray/commit/c74886a55e81dc8f4150094bfec69d875f0b365d to fix CI test for notebooks.

**Review by ericl — APPROVED** [review]

**Pull request merged**